zerver/views/users.py

```diff
@@ -389,6 +389,10 @@ def add_bot_backend(
                                  default_events_register_stream=default_events_register_stream,
                                  default_all_public_streams=default_all_public_streams,
                                  acting_user=user_profile)
+
+    if bot_type == UserProfile.ADMINISTRATOR_BOT and UserProfile.is_realm_admin:
+        do_change_user_role(bot_profile, UserProfile.ROLE_REALM_ADMINISTRATOR)
```

Comment:
I think we should instead check that the user is an administrator if that bot type is requested and error earlier in the function -- the `do_create_user` line should have created the user as an administrator.

Reply:
Yes the changes made in `check_bot_creation_policy` in `zerver/lib/users.py` take care of this.